src/main/scala/mesosphere/marathon/core/deployment/impl/TaskReplaceActor.scala

```diff
@@ -39,58 +39,80 @@ class TaskReplaceActor(
   // Killed resident tasks are not expunged from the instances list. Ignore
   // them. LaunchQueue takes care of launching instances against reservations
   // first
-  val currentRunningInstances = instanceTracker.specInstancesSync(runSpec.id).filter(_.isActive)
+  var currentRunningInstances: Seq[Instance] = _
 
   // In case previous master was abdicated while the deployment was still running we might have
   // already started some new tasks.
   // All already started and active tasks are filtered while the rest is considered
-  private[this] val (instancesAlreadyStarted, instancesToKill) = {
-    currentRunningInstances.partition(_.runSpecVersion == runSpec.version)
-  }
+  private[this] var instancesAlreadyStarted: Seq[Instance] = _
+  private[this] var instancesToKill: Seq[Instance] = _
 
   // The ignition strategy for this run specification
-  private[this] val ignitionStrategy = computeRestartStrategy(runSpec, currentRunningInstances.size)
+  private[this] lazy val ignitionStrategy: RestartStrategy = computeRestartStrategy(runSpec, currentRunningInstances.size)
```

Comment:
Maybe this should be a def? It seems problematic to have a lazy val encapsulate a value of a var the first time it is accessed.